docs/proposals/20220105-observatorium-alerting.md

```diff
@@ -0,0 +1,169 @@
+
+This API is different but a bit overlapping with `/api/metrics/v1/{tenant}/api/v1/rules/raw` which represents what was configured (not loaded).
+
+In order to fulfil the full story we also need:
+
+* Implement matching / filtering by label/tenant on Rules HTTP API of Querier.
+
+This will allow users looking at the Thanos Rules API (or Query UI) to only see the alerts that matter to them.
+
+### Alert Evaluation Status
+
+On top of mentioned Rules API, Prometheus also exposes [HTTP Alert API](https://prometheus.io/docs/prometheus/latest/querying/api/#alerts). We never expose it within Observatorium and Thanos and the truth is that Rules API gives exactly that information (on top of other data).
+
+Because of that we propose to not add this API, which will avoid a lot of work. In future iteration (e.g. if we would add UI), we could add Alert HTTP API in Observatorium which would extract data needed by this API from Rules API.
+
+### Alert Routing
+
+Triggered alerts are essential to be reliably forward to correct notifiers as soon as they are triggered. We propose deploying w-replica Alertmanager HA, through our configuration (without Prometheus Operator help which is usually used in our past deployments).
+
+We propose defining an additional HTTP API for routing on `/api/metrics/v1/{tenant}/api/v1/routing/raw` path in Observatorium API. Path that supports both GET, DELETE and PUT. 
+
+We propose to base the type on [Prometheus Operator AM Config](https://github.com/prometheus-operator/prometheus-operator/blob/9c0db5656f04e005de6a0413fd8eb8f11ec99757/pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go#L69). Proposed adding this API to upstream Alertmanager too (TODO).
+
+We could allow all read API as well of write of Inhibits and Silences of Alertmanager (https://github.com/prometheus/alertmanager/blob/main/api/v2/openapi.yaml) by adding auth and multi-tenancy layer, but we propose to deal with that in a separate epic/proposal.
+
+### Managing User Secrets
+
+Something new in Alertmanager Routing configuration is the fact that Alertmanager can integrate with other systemes ("receivers") to send alerts to. This requires TLS, basic auth or other type of authZ and authN to be configured within this configuration. AM supports it, but it expects secrets to be usually baked in file or available in local AM filesystem.
+
+This means we have to enter into user secret management story which is not trivial. We cannot just store those files in object storage in plain text. Encrypting them in rest sounds easy, but it involves lots of complexity around rotating keys, revoking secrets, process when leaked etc.
+
+We propose adding different backends depending on user preference to https://github.com/observatorium/config-client, but for a start we propose adding support to reading secrets via Vault. Unfortunately we have to delegate storing user credentials to 3rd party dependency. See alternatives for different ideas.
```

Comment:
TODO: Check if we AppSRE offers VaultaaS

Reply:
I have checked this for our internal purposes, and although AppSRE does not offer Vault for uses strictly outside of AppSRE Interface, Red Hat IAM services provide a Vault instance. We can discuss details internally.